### ISSUE TITLE
微调后模型推理和图片对不上

### ISSUE THREAD

**Issue by atomrun39** [THUDM/VisualGLM-6B#308]
`python cli_demo.py `，微调前描述和图像相符，微调后和图片不相符了
对于代码中提供的小狗图片示例，
微调前
![企业微信截图_16989145289256](https://github.com/THUDM/VisualGLM-6B/assets/69252333/b7d5e6d1-0e37-4450-b607-88f44963000e)
微调后
![企业微信截图_16989146962412](https://github.com/THUDM/VisualGLM-6B/assets/69252333/6f5f626e-6bb5-4124-b7a6-4c294716ded6)
微调脚本，我的训练数据有9w左右，是什么原因导致的呢：
`#! /bin/bash
NUM_WORKERS=1
NUM_GPUS_PER_WORKER=6
MP_SIZE=1

script_path=$(realpath $0)
script_dir=$(dirname $script_path)
main_dir=$(dirname $script_dir)
MODEL_TYPE="visualglm-6b"
MODEL_ARGS="--max_source_length 128 \
    --max_target_length 256 \
    --lora_rank 10 \
    --layer_range 0 14 \
    --pre_seq_len 4"

# OPTIONS_SAT="SAT_HOME=$1" #"SAT_HOME=/raid/dm/sat_models"
OPTIONS_NCCL="NCCL_DEBUG=info NCCL_IB_DISABLE=0 NCCL_NET_GDR_LEVEL=2"
HOST_FILE_PATH="hostfile"
HOST_FILE_PATH="hostfile_single"

train_data="/work/home/VisualGLM-6B-main/finetune_dataset/train_ft.json"
eval_data="/work/home/VisualGLM-6B-main/finetune_dataset/valid_ft.json"

gpt_options=" \
       --experiment-name finetune-$MODEL_TYPE \
       --model-parallel-size ${MP_SIZE} \
       --mode finetune \
       --train-iters 100000 \
       --resume-dataloader \
       $MODEL_ARGS \
       --train-data ${train_data} \
       --valid-data ${eval_data} \
       --distributed-backend nccl \
       --lr-decay-style cosine \
       --warmup .02 \
       --checkpoint-activations \
       --save-interval 10000 \
       --eval-interval 1000000 \
       --save "./checkpoints" \
       --split 1 \
       --eval-iters 15000 \
       --eval-batch-size 16 \
       --zero-stage 1 \
       --lr 0.0001 \
       --batch-size 16 \
       --skip-init \
       --fp16 \
       --use_lora
" 
run_cmd="${OPTIONS_NCCL} ${OPTIONS_SAT} deepspeed --master_port 16666 --include localhost:0,1,2,3,4,5 --hostfile ${HOST_FILE_PATH} updata_visualglm.py ${gpt_options}"
echo ${run_cmd}
eval ${run_cmd}

set +x`

**Comment by cdqncn** [comment]
我也是，微调的效果很差，甚至还有幻觉问题

**Comment by atomrun39** [comment]
求解这是为什么呀，我sat和VisualGLM-6B仓库代码都是新版本 @1049451037
跑官方例子，微调后直接乱码
![image](https://github.com/THUDM/VisualGLM-6B/assets/69252333/11ac15a1-dd4f-4e61-acdf-470d8853bb57)

**Comment by npzl** [comment]
同问，输入一个数据，输出别的数据的输出

**Comment by Xavier321456** [comment]
> 求解这是为什么呀，我sat和VisualGLM-6B仓库代码都是新版本 @1049451037 跑官方例子，微调后直接乱码 ![image](https://private-user-images.githubusercontent.com/69252333/280613765-11ac15a1-dd4f-4e61-acdf-470d8853bb57.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTA0ODU1NjAsIm5iZiI6MTcxMDQ4NTI2MCwicGF0aCI6Ii82OTI1MjMzMy8yODA2MTM3NjUtMTFhYzE1YTEtZGQ0Zi00ZTYxLWFjZGYtNDcwZDg4NTNiYjU3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMTUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzE1VDA2NDc0MFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTY0YjliN2EyYWNhODBhMDIxZDI5ZjMxN2I5YWEzNTAzZWY4NGU0NDI4NGEzYmJmNWY1MDcyYjBjODFjZDBhZGQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.g7vmDehh3CQPkJuOiH8RN06g21jwqnTYSOCXG1e_reA)

大哥我也遇到了这个问题，请问你最后解决了吗？

**Comment by corkiyao** [comment]
> > 求解这是为什么呀，我sat和VisualGLM-6B仓库代码都是新版本 @1049451037 跑官方例子，微调后直接乱码 ![image](https://private-user-images.githubusercontent.com/69252333/280613765-11ac15a1-dd4f-4e61-acdf-470d8853bb57.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTA0ODU1NjAsIm5iZiI6MTcxMDQ4NTI2MCwicGF0aCI6Ii82OTI1MjMzMy8yODA2MTM3NjUtMTFhYzE1YTEtZGQ0Zi00ZTYxLWFjZGYtNDcwZDg4NTNiYjU3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMTUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzE1VDA2NDc0MFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTY0YjliN2EyYWNhODBhMDIxZDI5ZjMxN2I5YWEzNTAzZWY4NGU0NDI4NGEzYmJmNWY1MDcyYjBjODFjZDBhZGQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.g7vmDehh3CQPkJuOiH8RN06g21jwqnTYSOCXG1e_reA)
> 
> 大哥我也遇到了这个问题，请问你最后解决了吗？

大哥，我也遇到这个问题了，请问解决了吗？
